src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -145,7 +145,7 @@ function getClauseForWorkspace(workspace: string) {
 
   return {
     bool: {
-      must: [{ term: { workspaces: workspace } }],
+      must: [{ term: { 'workspaces.keyword': workspace } }],
```

Comment:
the type of workspaces is `keyword`, do we need keyword here?

Reply:
The integration tests in `linux` will be failed, if we don't add the `.keyword` suffix.

Reply:
You can create `.kibana` index first with pre-defined mappings, otherwise openSearch will set the type automatically. the default type for string is `text` with sub field `keyword`, that's the case you have mentioned.

Reply:
Is there anyway to define this type in the saved objects side? It seems the `workspaces` field will be text type if mapping not defined. It could be a bug because we can't create `.kibana` index with pre-defined mapping when create saved objects in the real server.

Reply:
how about doing this in beforeAll method, you can replace mappings with real `.kibana` mappings
```
 await opensearchServer.opensearch.getClient().indices.create({
    body: { mappings, settings },
    index,
  });
```

Reply:
This only worked for integration tests right? Will we have the same problems in the real case?

Reply:
No, `IndexMigrator` will create `.kibana` with expected mapping when OSD startup.

Reply:
Fixed, feel free to help me re-review it.